### PR TITLE
R2_score for GLM

### LINKED ISSUE #2564: [Feature request] Bayesian Rsquare for GLM
[Gelman, Goodrich, Gabry, & Ali 2017](http://www.stat.columbia.edu/~gelman/research/unpublished/bayes_R2.pdf)

The concept of R^2 and “explained variance” makes most sense for GLM with equal variance, but it seems easy to compute so why not `¯\_(ツ)_/¯`

### PR DESCRIPTION
**Pull request by denadai2**
Fixes #2564.

Ref #2605.

### CODE REVIEW

File: pymc3/stats.py [hunk 1]
```diff
@@ -964,3 +964,33 @@ def bfmi(trace):
...
+
+    Returns
+    -------
+    `namedtuple` with the following elements:
+    R2_median: median of the Bayesian R2
+    R2_mean: mean of the Bayesian R2
+    R2_std: standard deviation of the Bayesian R2
+    """
+    e = y_true - y_pred
+    var_y_est = np.var(y_pred, 1)
```
Comment (aloctavodia) [inline pymc3/stats.py]: This is going to fail if `y_pred`has dimension 1.

File: pymc3/stats.py [hunk 2]
```diff
@@ -964,3 +964,33 @@ def bfmi(trace):
     energy = trace['energy']
 
     return np.square(np.diff(energy)).mean() / np.var(energy)
+
+
+def r2_score(y_true, y_pred):
```
Comment (aloctavodia) [inline pymc3/stats.py]: What about rounding the results by default (maybe 2 decimals) and adding an argument to control this...

File: pymc3/stats.py [hunk 3]
```diff
@@ -964,3 +964,33 @@ def bfmi(trace):
...
+    `namedtuple` with the following elements:
+    R2_median: median of the Bayesian R2
+    R2_mean: mean of the Bayesian R2
+    R2_std: standard deviation of the Bayesian R2
+    """
+    e = y_true - y_pred
+    var_y_est = np.var(y_pred, 1)
+    var_e = np.var(e, 1)
+
+    r2_median = np.median(var_y_est / (var_y_est + var_e))
```
Comment (aloctavodia) [inline pymc3/stats.py]: maybe you should assign `var_y_est / (var_y_est + var_e)` to a variable and then use that variable to compute the mean, median and std

**Comment by denadai2** [comment]
thx

**Comment by aloctavodia** [comment]
LGTM! In your previous PR you had a test, If possible you should add it again.

**Comment by aloctavodia** [comment]
Hi @denadai2 any update on this? Maybe we can skip the test, at least for now. What do you think @twiecki?

**Comment by denadai2** [comment]
@aloctavodia I didn't make any test in the previous pull request :/

**Comment by aloctavodia** [comment]
OK, so maybe no test for this PR :-)

**Comment by aloctavodia** [comment]
Thanks @denadai2

**Pull request merged**